examples/ownable/tests/ownable.rs

```diff
@@ -0,0 +1,173 @@
+#![cfg(feature = "e2e")]
+
+use abi::Ownable::OwnershipTransferred;
+use alloy::{
+    primitives::Address,
+    providers::Provider,
+    rpc::types::{BlockNumberOrTag, Filter},
+    sol,
+    sol_types::{SolConstructor, SolError, SolEvent},
+};
+use e2e::{receipt, send, EventExt, Revert, User};
+use eyre::Result;
+
+use crate::abi::Ownable;
+
+mod abi;
+
+sol!("src/constructor.sol");
+
+async fn deploy(user: &User, owner: Address) -> eyre::Result<Address> {
+    let args = OwnableExample::constructorCall { initialOwner: owner };
+    let args = alloy::hex::encode(args.abi_encode());
+    e2e::deploy(user.url(), &user.pk(), Some(args)).await
+}
+
+// ============================================================================
+// Integration Tests: Ownable
+// ============================================================================
+
+#[e2e::test]
+async fn constructs(alice: User) -> Result<()> {
+    let alice_addr = alice.address();
+    let contract_addr = deploy(&alice, alice_addr).await?;
+    let contract = Ownable::new(contract_addr, &alice.signer);
+
+    let Ownable::ownerReturn { owner } = contract.owner().call().await?;
+    assert_eq!(owner, alice_addr);
+
+    Ok(())
+}
+
+#[e2e::test]
+async fn emits_ownership_transfer_during_construction(
+    alice: User,
+) -> Result<()> {
+    let alice_addr = alice.address();
+
+    deploy(&alice, alice_addr).await?;
```

Comment:
Maybe you can try to get a tx receipt here? It would make event check much easier...

Reply:
I decided to do this first, because this needs changes in `koba`. I'll make it more ergonomic there and we can update here after.

Reply:
Okay, let's leave an issue and go with it.

Reply:
https://github.com/OpenZeppelin/rust-contracts-stylus/issues/145